src/main/java/io/fabric8/elasticsearch/plugin/kibana/KibanaSeed.java

```diff
@@ -318,11 +323,15 @@ private String getDefaultIndex(OpenshiftRequestContext context, Client esClient,
     }
 
     private void create(String kibanaIndex, List<String> projects, boolean setDefault, Client esClient,
-            String kibanaVersion, String projectPrefix) {
+            String kibanaVersion, String projectPrefix, Set<String> indexPatterns) {
         boolean defaultSet = !setDefault;
-
+        LOGGER.trace("Creating index-patterns for projects: '{}'", projects);
         for (String project : projects) {
-            createIndex(kibanaIndex, project, esClient, projectPrefix);
+            if(indexPatterns.contains(project)) { //no need to update
```

Comment:
What happens when we upgrade, and the new es image has new index patterns?  Do we have to manually delete the old index patterns first? I guess the real problem is when the user has created their own index patterns, but we need to add new ones during an upgrade.

Reply:
This is being driven by observations from @portante.  When the index patterns change, the only way to update them would be to delete the kibana index or the document in question.  The sole purpose of this PR is to avoid the issue seen by users where they build visualizations and dashboards based upon index mappings and then the mappings change.  The only solution I can thing of  would be to create an admin utility to allow updating mappings for a given user.  It might be easier to just delete there index.